src/Renderer/LayeredMaterial.js

```diff
@@ -398,6 +417,57 @@ LayeredMaterial.prototype.getTextureCountByLayerIndex = function getTextureCount
     return this.layerTexturesCount[index];
 };
 
+LayeredMaterial.prototype.getTextureByIndex = function getTextureByIndex(layerType, index) {
```

Comment:
`getTextureByIndex(l_COLOR, ...)` and `getTextureColorByIndex` are redundant from an API point of view.

`getTextureColorByIndex` shouldn't be exposed since its only a utility function.

And the same reasonning applies to all the new functions you're adding to LayeradMaterial: does it make sense to expose them?